solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -404,23 +404,26 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     rb.isDistrib = isDistrib(req);
     tagRequestWithRequestId(rb);
 
-    if (isShard) {
-      // log a simple message on start
-      log.info("Start Forwarded Search Query");
-      SolrParams filteredParams = removeVerboseParams(req.getParams());
-      rsp.getToLog()
-          .asShallowMap(false)
-          .put("params", "{" + filteredParams + "}"); // replace "params" with the filtered version
-    } else {
-      // Then it is the first time this req hitting Solr - not a req distributed by another higher
-      // level req.
-      // We have to log the query here as
-      // 1. It's useful to know the query before the processing start in case if the query stalls
-      // 2. The existing logging in SolrCore does not contain the query as query construction
-      // happens after the log
-      //    entries are added to rsp.toLog
-      if (log.isInfoEnabled()) {
-        log.info("Start External Search Query: {}", req.getParamString());
+    if (req.getPath() != null && req.getPath().contains("select")) {
+      if (isShard) {
+        // log a simple message on start
+        log.info("Start Forwarded Search Query");
+        SolrParams filteredParams = removeVerboseParams(req.getParams());
+        rsp.getToLog()
+            .asShallowMap(false)
+            .put(
+                "params", "{" + filteredParams + "}"); // replace "params" with the filtered version
+      } else {
```

Comment:
I'm wondering if instead of wrapping this in a check that the request is a "select" request, we should instead:
1. Change this else to be `else if (isDistrib(req))` to only log on distrib requests
2. Change the log level for these to debug so they don't log by default but we could enable if needed

Reply:
Today, we log `Start External Search Query` on query aggregator node, when alexa-request hits to QA node. Then we log  `Start Forwarded Search Query` on data node for each sub-shard request. Thus both if-else are relevant in context of `select` request.

if we want debug-logging for `get` request then need to find some other place as `RealTimeGetHandler` extends the `SearchHandler`.

Let me know if this looks reasonable